asm/f64/stubs_test.go

```diff
@@ -111,16 +120,118 @@ func TestCumProd(t *testing.T) {
 	} {
 		ret := CumProd(v.dst, v.src)
 		for i := range v.expect {
-			if ret[i] != v.expect[i] && (math.IsNaN(ret[i]) != math.IsNaN(v.expect[i])) {
+			if diff(ret[i], v.expect[i]) {
 				t.Log("Test", j, "CumProd error at", i, "Got:", ret[i], "Expected:", v.expect[i])
 				t.Fail()
 			}
-			if ret[i] != v.dst[i] && (math.IsNaN(ret[i]) != math.IsNaN(v.dst[i])) {
+			if diff(ret[i], v.dst[i]) {
 				t.Log("Test", j, "CumProd ret/dst mismatch", i, "Ret:", ret[i], "Dst:", v.dst[i])
 				t.Fail()
 			}
 		}
 	}
+	runtime.GC()
+}
+
+func TestDiv(t *testing.T) {
+	for j, v := range []struct {
+		dst, src, expect []float64
+	}{
+		{[]float64{1}, []float64{1}, []float64{1}},
+		{[]float64{nan}, []float64{nan}, []float64{nan}},
+		{[]float64{1, 2, 3, 4}, []float64{1, 2, 3, 4}, []float64{1, 1, 1, 1}},
+		{[]float64{2, 4, 6}, []float64{1, 2, 3, 4}, []float64{2, 2, 2}},
+		{[]float64{0, 0, 0, 0}, []float64{1, 2, 3}, []float64{0, 0, 0}},
+		{[]float64{nan, 1, nan, 1, 0},
+			[]float64{1, 1, nan, 1, 1},
+			[]float64{nan, 1, nan, 1, 0}},
+		{[]float64{inf, 4, nan, ninf, 9},
+			[]float64{inf, 4, nan, ninf, 3},
+			[]float64{nan, 1, nan, nan, 3}},
+	} {
+		Div(v.dst, v.src)
```

Comment:
I think that we may have been talking at crossed purposes.

I've just engineered a case that shows that what the `runtime.GC` call does is not what I would like tested; if I engineer the input to be corrupted (offset by 1 element - `[]float64{-inf, 4, nan, 8, 9, nan}[1:]`) and include a change in the ASM to counteract that but index into the slice outside (-1 element) the elements that I *should* be passing, the test passes. This is not a surprise to me, but it clearly shows that `runtime.GC` is not seeing unintended reads. If the slice is not offset with the same ASM change, then still don't get a throw (though we now fail as expected), but that absence of a throw is probably random depending on when/where things are allocated.